internal/manifestvalues/patch.go

```diff
@@ -17,6 +18,17 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+var resourceNameExpr = regexp.MustCompile("[^a-zA-Z0-9]+")
+
+func resourceName(s string) string {
+	return resourceNameExpr.ReplaceAllString(s, "-")
+}
```

Comment:
We already have something similar here: internal/names/utils.go would it make sense to use that instead? The difference would be of course, that we would get double dashes instead of single

Reply:
Oh nice, yes that should work, I will change it.

Reply:
With the existing `escapeResourceName` we do not get any dashes at all, because the points are not replaced. I am not sure if that is correct or not? 

If secret names are allowed to have '.' in them, we can use it (can't really tell from here https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)